### PR TITLE
Add 2.1 Dockerfiles

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related #398

### CODE REVIEW

**Comment by MichaelSimons** [comment]
@dotnet-bot test ci please

File: README.md [hunk 1]
```diff
@@ -10,26 +10,36 @@ See [dotnet/dotnet-docker](https://github.com/dotnet/dotnet-docker) for images w
...
+- [`2.0.0-runtime-jessie`, `2.0-runtime-jessie` (*2.0/runtime/jessie/amd64/Dockerfile*)](https://github.com/dotnet/dotnet-docker-nightly/blob/master/2.0/runtime/jessie/amd64/Dockerfile)
+- [`2.0.0-runtime-deps-stretch`, `2.0.0-runtime-deps`, `2.0-runtime-deps` (*2.0/runtime-deps/stretch/amd64/Dockerfile*)](https://github.com/dotnet/dotnet-docker-nightly/blob/master/2.0/runtime-deps/stretch/amd64/Dockerfile)
+- [`2.0.0-runtime-deps-jessie`, `2.0-runtime-deps-jessie` (*2.0/runtime-deps/jessie/amd64/Dockerfile*)](https://github.com/dotnet/dotnet-docker-nightly/blob/master/2.0/runtime-deps/jessie/amd64/Dockerfile)
+- [`2.0.0-sdk-stretch`, `2.0.0-sdk`, `2.0-sdk` (*2.0/sdk/stretch/amd64/Dockerfile*)](https://github.com/dotnet/dotnet-docker-nightly/blob/master/2.0/sdk/stretch/amd64/Dockerfile)
+- [`2.0.0-sdk-jessie`, `2.0-sdk-jessie` (*2.0/sdk/jessie/amd64/Dockerfile*)](https://github.com/dotnet/dotnet-docker-nightly/blob/master/2.0/sdk/jessie/amd64/Dockerfile)
+- [`2.1.0-runtime-stretch`, `2.1.0-runtime`, `2.1-runtime`, `2-runtime`, `runtime` (*2.1/runtime/stretch/amd64/Dockerfile*)](https://github.com/dotnet/dotnet-docker-nightly/blob/master/2.1/runtime/stretch/amd64/Dockerfile)
```
Comment (natemcmaster) [inline README.md]: Should we label these 2.1.0-preview1-* for now?

**Comment by natemcmaster** [comment]
By the way, https://github.com/aspnet/aspnet-docker/issues/281 makes me think we should be more careful with previews and the tags we give them.

**Comment by MichaelSimons** [comment]
@natemcmaster - agreed these should be tagged as 2.1.0-preview1-*.  That was an oversight on my part.  That being said, regarding "we should be more careful with previews", how are you suggesting we should be more careful?  In the past, to my recollection we have always indicated "preview" in our fully qualified tags of the official repos.  For example I don't think we used 2.0.0 prior to RTM.  Am I mistaken on this?  The nightly repos are a different story as we of course need to use the RTM tags prior to the actual RTM.

**Comment by natemcmaster** [comment]
> For example I don't think we used 2.0.0 prior to RTM.

Yeah, we're fine there. 

My concern is about the less-than-fully-qualified tags, such as `2.1`, and `2`. Maybe we should avoid adding these tags to previews. They are likely to break between previews. Once we RTM, the tags with `major.minor` should be fairly stable. In fact, these are the tags I recommend so you get patches automatically.

Side note: the `major` tag still leaves open the possibility of breaking when updates are published. A 2.0 app will not work on 2.1. The TFM, runtimes, runtime store, etc. all change too much between minor releases, but this is probably a conversation for another time.

**Comment by kendrahavens** [comment]
@natemcmaster Michael and I synced a little on this. We had understood that the 2.1 and 2 preview tags breaking when updated is actually a benefit so they will notify the CI machine (or user) when they are out of date. This is why we've used this pattern before.

**Comment by natemcmaster** [comment]
I think users would prefer to never break. IMO it would be better if we didn't publish `major.minor` until RTM, and dropped `major` altogether.

**Comment by MichaelSimons** [comment]
When settling on the Docker tagging strategy for .NET Core, the discussions about the `major.minor` and `major` tags centered around developer scenarios.  Imagine a CI system setup to always build on these tags as a way to detect breaks.

That being said, we are certainly open to feedback from the community on the tags we produce.  Having not received any community feedback thus far, I am reluctant to make changes.

**Comment by MichaelSimons** [comment]
Going to merge in order to get 2.1 images.  If there are are further suggestions/imput on tagging please open an issue.  We have time to make changes before the 2.1 images are migrated to microsoft/dotnet.

**Pull request merged**